lib/Wikia/src/Persistence/User/Preferences/PreferencePersistence.php

```diff
@@ -228,9 +232,39 @@ private function findWikisWithLocalPreference( ReverseLookupApi $api, $preferenc
 		return $wikiList;
 	}
 
-	private function findUsersWithGlobalPreference( ReverseLookupApi $api, $preferenceName, $value = null ) {
-		$userList = $api->findUsersWithGlobalPreference( $preferenceName, $value );
-		return $userList;
+	/**
+	 * Calls user-preference /reverse-lookup/global/{preferenceName}/users endpoint to get an array of string userIds
+	 * of users with $preferenceName set to $value (or set at all if $value is null)
+	 * Handles paginating API calls for $numUsersToRetrieve > 10000
+	 * @param ReverseLookupApi $api
+	 * @param string $preferenceName
+	 * @param string|null $value
+	 * @param int|null $numUsersToRetrieve
+	 * @param int|null $userIdContinue
+	 * @return array|string[] - userId
+	 * @throws ApiException
+	 */
+	private function findUsersWithGlobalPreference(ReverseLookupApi $api, $preferenceName, $value = null, $numUsersToRetrieve = 1000, $userIdContinue = null ) {
+		if ( $numUsersToRetrieve <= PreferencePersistence::REVERSE_LOOKUP_GLOBAL_USERS_MAX_LIMIT ) {
+			return $api->findUsersWithGlobalPreference( $preferenceName, $value, $numUsersToRetrieve, $userIdContinue );
+		}
+		else {
+			// pagination for API calls
+			$userList = [];
```

Comment:
Are you sure that the number of users to retrieve does not need to be limited?

What if someone retrieves 2M of users? The code will try to add all of them to the userList collection and probably run out of memory.